generated `bindings` error, with missing `mut` on `arg1`

Given deps
```toml
[dependencies]
bitflags = "2.4.2"
wit-bindgen-rt = "0.21.0"
```
and `wit` including

```wit
render: func(ctx: context) -> result<string, string>;
```

Generates the `fn` signature as:

```rust
pub unsafe fn _export_render_cabi<T: Guest>(
                    arg0: i32,
                    arg1: ::core::mem::MaybeUninit<u64>,
                    arg2: i64,
                    arg3: i32,
                    arg4: f64,
                    arg5: i32,
                    arg6: *mut u8,
                    arg7: usize,
                ) -> *mut u8
```

but `arg1` needs to be `mut` according to the eror:

```
error[E0596]: cannot borrow `arg1` as mutable, as it is not declared as mutable
   --> src/bindings.rs:279:37
    |
279 | ...                   arg1.as_mut_ptr().cast::<*mut u8>().read().cast(),
    |                       ^^^^ cannot borrow as mutable
    |
help: consider changing this to be mutable
    |
222 |                     mut arg1: ::core::mem::MaybeUninit<u64>,
    |                     +++

```

<details>
<summary>Full code</summary>

```rust
                #[doc(hidden)]
                #[allow(non_snake_case)]
                pub unsafe fn _export_render_cabi<T: Guest>(
                    arg0: i32,
                    arg1: ::core::mem::MaybeUninit<u64>,
                    arg2: i64,
                    arg3: i32,
                    arg4: f64,
                    arg5: i32,
                    arg6: *mut u8,
                    arg7: usize,
                ) -> *mut u8 {
                    use super::super::super::super::component::test_wit_newest::context_types::Context as V2;
                    let v2 = match arg0 {
                        0 => {
                            let e2 = super::super::super::super::component::test_wit_newest::context_types::Content{
            revenue: match arg1.assume_init() as i64 as i32 {
              0 => None,
              1 => {
                let e = f64::from_bits(arg2 as u64);
                Some(e)
              }
              _ => _rt::invalid_enum_discriminant(),
            },
            expenses: match arg3 {
              0 => None,
              1 => {
                let e = arg4;
                Some(e)
              }
              _ => _rt::invalid_enum_discriminant(),
            },
            formula: match arg5 {
              0 => None,
              1 => {
                let e = {
                  let len0 = arg7;
                  let bytes0 = _rt::Vec::from_raw_parts(arg6.cast(), len0, len0);

                  _rt::string_lift(bytes0)
                };
                Some(e)
              }
              _ => _rt::invalid_enum_discriminant(),
            },
          };
                            V2::AllContent(e2)
                        }
                        1 => {
                            let e2 = f64::from_bits(arg1.assume_init() as i64 as u64);
                            V2::Revenue(e2)
                        }
                        2 => {
                            let e2 = f64::from_bits(arg1.assume_init() as i64 as u64);
                            V2::Expenses(e2)
                        }
                        n => {
                            debug_assert_eq!(n, 3, "invalid enum discriminant");
                            let e2 = {
                                let len1 = arg2 as usize;
                                let bytes1 = _rt::Vec::from_raw_parts(
                                    arg1.as_mut_ptr().cast::<*mut u8>().read().cast(),
                                    len1,
                                    len1,
                                );

                                _rt::string_lift(bytes1)
                            };
                            V2::Formula(e2)
                        }
                    };
```
</details>

--- a/crates/rust/src/lib.rs
+++ b/crates/rust/src/lib.rs
@@ -1357,7 +1357,7 @@ fn perform_cast(operand: &str, cast: &Bitcast) -> String {
         // Convert a `MaybeUninit<u64>` holding a pointer value back into
         // the pointer value.
         Bitcast::P64ToP => {
-            format!("{}.as_mut_ptr().cast::<*mut u8>().read()", operand)
+            format!("{}.as_ptr().cast::<*mut u8>().read()", operand)
         }
         // Convert an `i32` or a `usize` into a pointer.
         Bitcast::I32ToP | Bitcast::LToP => {

--- a/tests/codegen/variants-unioning-types.wit
+++ b/tests/codegen/variants-unioning-types.wit
@@ -92,4 +92,10 @@ world a {
     b(f32),
   }
   import f-s64-f32: func(x: v-s64-f32);
+
+  variant v-string-u64 {
+    a(string),
+    b(u64),
+  }
+  export f-string-u64: func(x: v-string-u64);
 }

